x-pack/plugins/observability_solution/entity_manager/server/lib/entities/built_in/services.ts

```diff
@@ -8,63 +8,120 @@
+};
+
+export const builtInServicesFromLogsEntityDefinition: EntityDefinition =
+  entityDefinitionSchema.parse({
+    version: '0.1.0',
+    id: `${BUILT_IN_ID_PREFIX}services_from_ecs_data`,
+    name: 'Services from ECS data',
+    description:
+      'This definition extracts service entities from common data streams by looking for the ECS field service.name',
+    type: 'service',
+    managed: true,
+    filter: '@timestamp >= now-10m',
+    indexPatterns: ['logs-*', 'filebeat*', 'metrics-apm*'],
+    history: {
+      timestampField: '@timestamp',
+      interval: '1m',
+      settings: {
+        frequency: '2m',
+        syncDelay: '2m',
+      },
     },
-    {
-      name: 'errorRate',
-      equation: 'A / 5',
-      metrics: [
-        {
-          name: 'A',
-          aggregation: 'doc_count',
-          filter: 'log.level: "ERROR"',
-        },
-      ],
+    latest: {
+      lookback: '5m',
     },
-  ],
-});
+    identityFields: ['service.name', { field: 'service.environment', optional: true }],
+    displayNameTemplate: '{{service.name}}{{#service.environment}}:{{.}}{{/service.environment}}',
+    metadata: [
+      { source: '_index', destination: 'sourceIndex' },
+      'data_stream.type',
+      'service.environment',
+      'service.name',
+      'service.namespace',
+      'service.version',
+      'service.runtime.name',
+      'service.runtime.version',
+      'service.language.name',
+      'cloud.provider',
+      'cloud.availability_zone',
+      'cloud.machine.type',
+    ],
+    metrics: [
+      {
+        name: 'latency',
+        equation: 'A',
+        metrics: [
+          {
+            name: 'A',
+            aggregation: 'avg',
+            filter: serviceTransactionFilter(),
+            field: 'transaction.duration.histogram',
+          },
+        ],
+      },
+      {
+        name: 'throughput',
+        equation: 'A',
+        metrics: [
+          {
+            name: 'A',
+            aggregation: 'doc_count',
+            filter: serviceTransactionFilter(),
+          },
+        ],
+      },
+      {
+        name: 'failedTransactionRate',
+        equation: 'A / B',
+        metrics: [
+          {
+            name: 'A',
+            aggregation: 'doc_count',
+            filter: serviceTransactionFilter(['event.outcome: "failure"']),
+          },
+          {
+            name: 'B',
+            aggregation: 'doc_count',
+            filter: serviceTransactionFilter(['event.outcome: *']),
+          },
+        ],
+      },
+      {
+        name: 'logErrorRate',
+        equation: 'A / B',
+        metrics: [
+          {
+            name: 'A',
+            aggregation: 'doc_count',
+            filter: 'log.level: "error" OR error.log.level: "error"',
+          },
+          {
+            name: 'B',
+            aggregation: 'doc_count',
+            filter: 'log.level: * OR error.log.level: *',
+          },
+        ],
+      },
+      {
+        name: 'logRate',
```

Comment:
Does this track the `logRate` per minute? If so, can we rename it to make it more clear? ex logRatePerMinute

Reply:
Every aggregation would be per minute here that's why I didn't specify it, do you think we should add the suffix to all ?

Reply:
I'd vote for yes, we had some discussion about this in the API guidelines document.
Adding the unit aligns more with how ECS names fields and makes it clearer for the consumer what to expect.
It also means that if we want to change the unit we need to consider that a breaking change (or add a new field in a different unit), which is correct since the users might do further computation assuming the unit is correct.

Reply:
I'm not convinced from a dx standpoint, if we update the definition to aggregate over different periods or if we introduce different aggregation (ie 1m, 5m, 30m) the code relying on these aggregations would quite suffer. I'd rather add a different field that would specify the metrics periods like `metricset.period`

Reply:
Can you expand a bit on how the relying could would be affected by such a change? I can't see how that would be a heavy downstream change but sure, compared to using a computed value anything seems more but I think the value is that it's more clear for the consumers (end user or Kibana apps) and I'd favor that over our own DX.

Reply:
Code needs to change because we introduce new fields. it's just redundant information